tests/unit_tests/dft/source/descriptor_tests.cpp

```diff
@@ -405,6 +409,106 @@ inline void set_readonly_values(sycl::queue& sycl_queue) {
+          std::make_pair(config_param::FORWARD_SCALE, PrecisionType{ 1.2 }),
+          std::make_pair(config_param::BACKWARD_SCALE, PrecisionType{ 3.4 }) },
+        { std::make_pair(config_param::NUMBER_OF_TRANSFORMS, std::int64_t{ 5 }),
+          std::make_pair(config_param::COMPLEX_STORAGE, config_value::COMPLEX_COMPLEX),
+          std::make_pair(config_param::REAL_STORAGE, config_value::REAL_REAL),
+          std::make_pair(config_param::CONJUGATE_EVEN_STORAGE, config_value::COMPLEX_COMPLEX) },
+        { std::make_pair(config_param::PLACEMENT, config_value::NOT_INPLACE),
+          std::make_pair(config_param::INPUT_STRIDES, strides.data()),
+          std::make_pair(config_param::OUTPUT_STRIDES, strides.data()),
+          std::make_pair(config_param::FWD_DISTANCE, std::int64_t{ 60 }),
+          std::make_pair(config_param::BWD_DISTANCE, std::int64_t{ 70 }) },
+        { std::make_pair(config_param::WORKSPACE, config_value::ALLOW),
+          std::make_pair(config_param::ORDERING, config_value::ORDERED),
+          std::make_pair(config_param::TRANSPOSE, bool{ false }),
+          std::make_pair(config_param::PACKED_FORMAT, config_value::CCE_FORMAT) }
+    };
+
+    for (int i = 0; i < argument_groups.size(); i += 1) {
+        for (auto argument : argument_groups[i]) {
+            std::visit([&descriptor, p = argument.first](auto&& a) { descriptor.set_value(p, a); },
+                       argument.second);
+        }
+        try {
+            commit_descriptor(descriptor, sycl_queue);
+        }
+        catch (oneapi::mkl::exception& e) {
+            FAIL() << "exception at index " << i << " with error : " << e.what();
+        }
+    }
+}
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+inline void change_queue_causes_wait(sycl::queue& busy_queue) {
+    // create a queue with work on it, and then show that work is waited on when the descriptor
+    // is committed to a new queue.
+    // its possible to have a false positive result, but a false negative should not be possible.
+    // sleeps have been added to reduce the false positives to show that we are actually waiting for
+    // notification/queue.
+    using namespace std::chrono_literals;
+    std::condition_variable cv;
+    std::mutex cv_m;
+    // signal used to avoid spurious wakeups
+    bool signal = false;
+
+    sycl::queue free_queue;
+
+    // commit the descriptor on the "busy" queue
+    oneapi::mkl::dft::descriptor<precision, domain> descriptor{ default_1d_lengths };
+    EXPECT_NO_THROW(commit_descriptor(descriptor, busy_queue));
+
+    // add some work to the busy queue
+    auto e = busy_queue.submit([&](sycl::handler& cgh) {
+        cgh.host_task([&] {
+            std::unique_lock<std::mutex> lock(cv_m);
+            ASSERT_TRUE(cv.wait_for(lock, 5s, [&] { return signal; })); // returns false on timeout
+            std::this_thread::sleep_for(500ms);
+        });
+    });
+    std::this_thread::sleep_for(500ms);
```

Comment:
I don't follow how this sleep and the one above reduce the false positives?
I'm not really concerned by this anyway so I'm approving again.

Reply:
I am generally against using sleeps in tests, but I hope these make sense.
This sleep gives the above event a chance to complete, semi-proving that it will have at least started. I didn't want to assert that is was in-progress because that could technically fail.
The one above makes the kernel last longer because the conditional variable will quite likely get the mutex before the commit begins, so the kernel may finish before the commit even started.

Potentially making both sleeps the same length ruins the effect of the first one...

Reply:
Thanks I got it now.